src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseCreationResult.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.events.CaseCreationResultType;
+
+public final class CaseCreationResult {
+
+    public final CaseCreationResultType resultType;
+    public final Long caseCcdId;
+
+    private CaseCreationResult(CaseCreationResultType resultType, Long caseCcdId) {
+        this.resultType = resultType;
+        this.caseCcdId = caseCcdId;
+    }
+
+    public static CaseCreationResult createdCase(Long caseCcdId) {
+        return new CaseCreationResult(CaseCreationResultType.CREATED_CASE, caseCcdId);
+    }
+
+    public static CaseCreationResult caseAlreadyExists(Long caseCcdId) {
+        return new CaseCreationResult(CaseCreationResultType.CASE_ALREADY_EXISTS, caseCcdId);
+    }
+
+    public static CaseCreationResult unrecoverableFailure() {
+        return new CaseCreationResult(CaseCreationResultType.UNRECOVERABLE_FAILURE, null);
+    }
+
+    public static CaseCreationResult potentiallyRecoverableFailure() {
+        return new CaseCreationResult(CaseCreationResultType.POTENTIALLY_RECOVERABLE_FAILURE, null);
+    }
+
+    public static CaseCreationResult abortedWithoutFailure() {
+        return new CaseCreationResult(CaseCreationResultType.ABORTED_WITHOUT_FAILURE, null);
+    }
+}
```

Comment:
new CaseCreationResult(CaseCreationResultType.UNRECOVERABLE_FAILURE, null),
new CaseCreationResult(CaseCreationResultType. POTENTIALLY_RECOVERABLE_FAILURE, null),
new CaseCreationResult(CaseCreationResultType. ABORTED_WITHOUT_FAILURE, null)
can be final static (i.e. constants) created just once

Reply:
done